main.go

```diff
@@ -0,0 +1,205 @@
+	if err != nil {
+		return nil, err
+	}
+	return domains, nil
+}
+
+func findDomain(cfc *client.Client, query string) (*resource.Domain, string, *url.URL, error) {
+	routeUrl, err := url.Parse(query)
+	if err != nil {
+		return &resource.Domain{}, "", &url.URL{}, err
+	}
+	if routeUrl.Scheme == "" {
+		return &resource.Domain{}, "", routeUrl, fmt.Errorf("please provide the url including the scheme")
+	}
+
+	domains, err := retrieveDomains(cfc, routeUrl.Hostname())
+	if err != nil {
+		return &resource.Domain{}, routeUrl.Hostname(), routeUrl, err
+	}
+
+	if len(domains) == 0 {
+		urlParts := strings.SplitN(routeUrl.Hostname(), ".", 2)
+		if len(urlParts) < 2 {
+			return &resource.Domain{}, "", routeUrl, fmt.Errorf("'%s' is not a domain", routeUrl.Hostname())
+		}
+		hostName := urlParts[0] //Subdomain is not empty
+		domainName := urlParts[1]
+		domains, err = retrieveDomains(cfc, domainName)
+		if len(domains) == 0 {
+			return &resource.Domain{}, hostName, routeUrl, fmt.Errorf("error retrieving apps: route not found, domain '%s' is unknown", domainName)
+		}
+		return domains[0], hostName, routeUrl, nil
+	}
+	return domains[0], routeUrl.Hostname(), routeUrl, nil
+}
+
+func findRoute(cfc *client.Client, query string) (*resource.Route, error) {
+	domain, hostName, routeUrl, err := findDomain(cfc, query)
+
+	opts := client.NewRouteListOptions()
+	opts.Hosts.Values = append(opts.Hosts.Values, hostName)
+	opts.DomainGUIDs.Values = append(opts.DomainGUIDs.Values, domain.GUID)
+	opts.Paths.Values = append(opts.Paths.Values, routeUrl.Path)
+
+	routes, err := cfc.Routes.ListAll(context.Background(), opts)
+	if err != nil {
+		return &resource.Route{}, err
+	}
+
+	if len(routes) == 0 { // Wildcard search
+		opts.Hosts.Values = append(opts.Hosts.Values, "*")
+		routes, err = cfc.Routes.ListAll(context.Background(), opts)
+		if err != nil {
+			return &resource.Route{}, err
+		}
+		if len(routes) == 0 {
+			return &resource.Route{}, fmt.Errorf("error retrieving apps: route '%s' not found", routeUrl.Hostname())
+		}
+	}
+
+	return routes[0], nil
+}
+
+func lookup(cfc *client.Client, route *resource.Route, target bool, cliConnection plugin.CliConnection) error {
+	if route.Destinations == nil || len(route.Destinations) == 0 {
+		return fmt.Errorf("route not bound to any applications")
+	}
+
+	destinationCount := 0
+	for _, destination := range route.Destinations {
+		app, err := cfc.Applications.Get(context.Background(), *destination.App.GUID)
+		if err != nil {
+			return fmt.Errorf("route not bound to any applications")
+		}
+
+		space, org, err := cfc.Spaces.GetIncludeOrganization(context.Background(), app.Relationships.Space.Data.GUID)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("Bound to:\nOrganization: %s (%s)\n", org.Name, org.GUID)
+		fmt.Printf("Space       : %s (%s)\n", space.Name, space.GUID)
+		fmt.Printf("App         : %s (%s)\n", app.Name, app.GUID)
+
+		if target && destinationCount == 0 {
```

Comment:
It would be good if this was happening _after_ all the apps were listed.

Then you have a clean output:
```
Bound to:
Organization: my cool org
Space: my even cooler space
App: gottagofast (1234-54642-124CAFE)
App: sonicthehedgehog (9274-A41947-d423a)

Targeting my cool org / my even cooler space ...
$(cf target -o "my cool org" -s "my even cooler space")
```

Reply:
done